devito/symbolics/queries.py

```diff
@@ -50,16 +50,19 @@ def q_trigonometry(expr):
     return expr.is_Function and expr.func in [sin, cos]
 
 
-def q_op(expr):
-    return expr.is_Add or expr.is_Mul or expr.is_Function
+def q_routine(expr):
+    from devito.types.basic import AbstractFunction
+    return expr.is_Function and not isinstance(expr, AbstractFunction)
 
 
 def q_xop(expr):
-    return q_op(expr) or expr.is_Pow
+    return (expr.is_Add or expr.is_Mul or expr.is_Pow or q_routine(expr))
```

Comment:
`is_Mod`?

Reply:
don't expect to ever find it in compute intensive expressions ?
and if it did, I wouldn't know how the caller would treat this information